lib/components/user/after-signin-switch.js

```diff
@@ -0,0 +1,49 @@
+import { replace } from 'connected-react-router'
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+
+import { routeTo } from '../../actions/ui'
+import { isNewUser } from '../../actions/user'
+
+import AfterSignInScreen from './after-signin-screen'
+import withLoggedInUserSupport from './with-logged-in-user-support'
+
+/**
+ * This screen is flashed just after user sign in while state.user.loggedInUser is fetched.
+ * Once state.user.loggedInUser is available:
+ * - For new users, route them to the account page (it will show account setup).
+ * - For existing users, route them to the itinerary search that was in place before login.
+ */
+class AfterSignInSwitch extends Component {
+  componentDidUpdate () {
+    const { loggedInUser, routeTo } = this.props
+
+    // It will take several calls to componentDidUpdate() before loggedInUser becomes available.
+    if (loggedInUser) {
+      const nextPath = isNewUser(loggedInUser) ? '/account' : '/'
+
+      // Reroute and prevent the current URL from showing in the browser history.
+      routeTo(nextPath, null, replace)
```

Comment:
I still think this logic would be better contained in an action. Also, this component should be merged with `AfterSignInScreen`.

Reply:
Also, if the user is not new, we should probably be redirecting to whatever path they were currently at, right? Not just `/`.

Reply:
Ideally.

Reply:
Resolved with commit https://github.com/opentripplanner/otp-react-redux/pull/167/commits/8670c7b266bf77a090309d4b15a19f2f6644e4eb.